src/main/java/org/embulk/output/mailchimp/MailChimpClient.java

```diff
@@ -115,90 +116,103 @@ public void handleErrors(List<ErrorResponse> errorResponses)
-                        page++;
-                    }
-                    else {
-                        hasMore = false;
-                    }
-                }
-
-                Function<CategoriesResponse, String> function = new Function<CategoriesResponse, String>()
-                {
-                    @Override
-                    public String apply(CategoriesResponse input)
-                    {
-                        return input.getTitle().toLowerCase();
-                    }
-                };
-
-                // Transform to a list of available category names and validate with data that user input
-                ImmutableList<String> availableCategories = FluentIterable
-                        .from(allCategoriesResponse)
-                        .transform(function)
-                        .toList();
-
-                for (String category : taskGroupingColumns) {
-                    if (!availableCategories.contains(category)) {
-                        throw new ConfigException("Invalid interest category name: '" + category + "'");
-                    }
+            List<Category> categories = fetchCategories(retryable, task.getListId(), task.getGroupingColumns().get());
+            Map<String, Map<String, Interest>> interestsByCategory = new HashMap<>();
+            for (Category category : categories) {
+                // Skip fetching interests if this category isn't specified in the task's grouping column.
+                // Assume task's grouping columns are always in lower case
+                if (!task.getGroupingColumns().get().contains(category.getTitle().toLowerCase())) {
+                    continue;
                 }
+                avoidFloodAPI("Fetching next category's interests", task.getSleepBetweenRequestsMillis());
+                interestsByCategory.put(
+                        category.getTitle().toLowerCase(),
+                        convertInterestCategoryToMap(fetchInterests(retryable, task.getListId(), category.getId())));
+            }
 
-                for (CategoriesResponse categoriesResponse : allCategoriesResponse) {
-                    // Skip fetching interests if this category isn't specified in the task's grouping column.
-                    // Assume task's grouping columns are always in lower case (like `availableCategories` did)
-                    if (!taskGroupingColumns.contains(categoriesResponse.getTitle().toLowerCase())) {
-                        continue;
-                    }
-
-                    String detailPath = MessageFormat.format("/lists/{0}/interest-categories/{1}/interests",
-                                                             task.getListId(),
-                                                             categoriesResponse.getId());
-                    response = jsonParser.parseJsonObject(retryable.get(detailPath));
+            // Warn if schema doesn't have the task's grouping column
```

Comment:
Since this check doesn't need extra data or involve in client/data should we move this to the `validateOutputTask` method instead?

Reply:
Good idea